### ISSUE TITLE
Which characters are considered valid?

### ISSUE THREAD

**Issue by NotUnlikeTheWaves** [dakyskye/dxhd#48]
I'm especially thinking towards control characters. For example, in a range `{0-9}` you have the control characters {, -, and } that a user might also want to use as part of a binding. For example, maybe someone might want to minimize a window with `shift + -`.

Do we consider control characters also as valid 'regular' characters that can be part of a binding?

If so, should we prefix them with an escape character perhaps? That is, `shift + \-`.

The only place I can think of where an ambiguity can arise is in the case of the comma `,` as the separator in an option list (`{a, b}`).

**Comment by dakyskye** [comment]
@NotUnlikeTheWaves we'll need to support escaping of characters anyways, because a user might want to echo "{" symbol. We definitely don't want our parser to think it's the beginning of a variants group.

**Comment by NotUnlikeTheWaves** [comment]
Good point. We need to explore how different the binding parsing is vs the action substitution
